iroh-net/src/discovery/pkarr/dht.rs

```diff
@@ -0,0 +1,365 @@
+//! # Pkarr based node discovery for iroh-net
```

Comment:
this has no tests, would be good to have some basic test to make sure this works

Reply:
I will see what I can do. The mainline crate has a testnet feature, otherwise I would have to assume that the global DHT is available from tests...